ios/bottom-tabs/RNSBottomAccessoryHelper.mm

```diff
@@ -0,0 +1,194 @@
+
+- (id<UITraitChangeRegistration>)registerForAccessoryEnvironmentChanges
+{
+  return [_bottomAccessoryView
+      registerForTraitChanges:@[ [UITraitTabAccessoryEnvironment class] ]
+                  withHandler:^(__kindof id<UITraitEnvironment>, UITraitCollection *previousTraitCollection) {
+                    [self->_bottomAccessoryView.reactEventEmitter
+                        emitOnEnvironmentChangeIfNecessary:self->_bottomAccessoryView.traitCollection
+                                                               .tabAccessoryEnvironment];
+                  }];
+}
+
+#pragma mark - Observing frame changes
+
+- (void)registerForAccessoryFrameChanges
+{
+  [self.nativeWrapperView addObserver:self forKeyPath:@"center" options:NSKeyValueObservingOptionInitial context:nil];
+}
+
+- (void)observeValueForKeyPath:(NSString *)keyPath
+                      ofObject:(id)object
+                        change:(NSDictionary *)change
+                       context:(void *)context
+{
+  [self notifyFrameUpdate];
+}
+
+- (UIView *)nativeWrapperView
+{
+  RCTAssert(
+      _bottomAccessoryView.superview.superview != nil,
+      @"[RNScreens] RNSBottomTabsAccessoryComponentView must be the set as bottom accessory.");
+  return _bottomAccessoryView.superview.superview;
+}
+
+- (void)notifyFrameUpdate
+{
+#if !RCT_NEW_ARCH_ENABLED || REACT_NATIVE_VERSION_MINOR < 82
+  // Make sure that bottom accessory's size is sent to ShadowNode as soon as possible.
+  // We set origin to (0,0) because initially self.nativeWrapperView's origin is incorrect.
+  // We want the enable the display link as well so that it takes over later with correct origin.
+  if (!_initialStateUpdateSent) {
+    CGRect frame = CGRectMake(0, 0, self.nativeWrapperView.frame.size.width, self.nativeWrapperView.frame.size.height);
+    [self updateShadowStateWithFrame:frame];
+    _initialStateUpdateSent = YES;
+  }
+
+  if (_displayLink == nil && !CGRectEqualToRect(_previousFrame, self.nativeWrapperView.frame)) {
+    [self setupDisplayLink];
+  }
+#else // !RCT_NEW_ARCH_ENABLED || REACT_NATIVE_VERSION_MINOR < 82
+  // We use self.nativeWrapperView because it has both the size and the origin
+  // that we want to send to the ShadowNode.
+  [self updateShadowStateWithFrame:self.nativeWrapperView.frame];
+#endif // !RCT_NEW_ARCH_ENABLED || REACT_NATIVE_VERSION_MINOR < 82
+}
+
+#if !RCT_NEW_ARCH_ENABLED || REACT_NATIVE_VERSION_MINOR < 82
+- (void)setupDisplayLink
+{
+  _displayLink = [CADisplayLink displayLinkWithTarget:self selector:@selector(handleDisplayLink:)];
+  [_displayLink addToRunLoop:[NSRunLoop mainRunLoop] forMode:NSRunLoopCommonModes];
+}
+
+- (void)handleDisplayLink:(CADisplayLink *)sender
+{
+  // We use self.nativeWrapperView because it has both the size and the origin
+  // that we want to send to the ShadowNode.
+  CGRect presentationFrame = self.nativeWrapperView.layer.presentationLayer.frame;
+  if (CGRectEqualToRect(presentationFrame, CGRectZero)) {
+    return;
+  }
```

Comment:
When can it be CGRectZero? Only before the very first layout or do you handle some other cases here?